gym/envs/registration.py

```diff
@@ -305,24 +435,26 @@ def make(self, path: str, **kwargs) -> Env:
             logger.info("Making new env: %s", path)
         spec = self.spec(path)
 
-        # Match the parts of the environment ID as we need to parse
-        # if there's a newer version of this environment.
-        match = env_id_re.fullmatch(spec.id)
-        assert match is not None  # Can't be hit as self.spec checks
-        namespace, name, version = match.group("namespace", "name", "version")
-
-        # Get all versions in the requested namespace.
-        versions = self._versions(namespace, name)
+        # Get all versions of this spec.
+        versions = self.env_specs.versions(spec.namespace, spec.name)
 
         # We check what the latest version of the environment is and display a warning
         # if the user is attempting to initialize an older version.
-        latest_version, latest_ns = max(versions)
-        if int(version) < latest_version:
-            latest_id = env_id_from_parts(latest_ns, name, latest_version)
+        latest_spec = max(
```

Comment:
Maybe rename this to `latest_versioned_spec` so it's clear we're filtering the default out.